### PR TITLE
Bookstore UI

### PR DESCRIPTION
**Pull request by george-swift**
### Objectives
In this milestone, the components are redesigned to match the design of the user interface on [Zeplin](https://app.zeplin.io/project/5b35a9e13227086040f8eb75/screen/5b695e29bb8c844f118f9378).

The following changes were made:
- A `Header` presentational component is added
- `BooksList` and `Book` components are redesigned from table elements to match UI
- Styling rules from the design file are adhered to and each element looks exactly the same as in the design.

### Additional Info
- Placeholders are added to unavailable functionality. The authors of the original books in the store are added to the named export file. However, adding a new book will only display `Author's Placeholder`. The progress and chapters are random numbers.
- **Only after** meeting the design specs to the best of my ability, media queries were added for smaller screen sizes.

### CODE REVIEW

File: src/assets/Sass/_mediaqueries.scss [hunk 1]
```diff
@@ -0,0 +1,183 @@
+@media screen and (max-width: 1200px) {
+  .header {
+    padding-left: 3.09rem;
```
Comment (e71az) [inline src/assets/Sass/_mediaqueries.scss:3]: 3.09rem is repeated 6 times, please create classes for this kind of variable cases.

File: src/assets/Css/App.css [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+  min-width: 15rem;
+  height: 2.313rem;
+  margin: 0.375rem 2.938rem 0.125rem 0;
+  color: #0290ff;
+  font: bold 1.875rem "Montserrat", sans-serif;
+  text-decoration: none;
+  letter-spacing: normal;
+  font-stretch: normal;
+  font-style: normal;
+  line-height: normal;
```
Comment (e71az) [inline src/assets/Css/App.css:38]: These rules are repeated 10 times.

File: src/assets/Css/App.css [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+  width: 6.563rem;
+  margin: 1.125rem 0 0.688rem 0;
+  opacity: 0.5;
+}
+
+#books,
+#categories {
+  height: 1rem;
+  font: normal 0.813rem/1.9px "Montserrat", sans-serif;
+  color: #121212;
```
Comment (e71az) [inline src/assets/Css/App.css:69]: You created a class for this value but forgot to use it in 3 cases.

File: src/assets/Css/App.css [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+
+.book-row:last-child {
+  margin-bottom: 0;
+}
+
+.book-name {
+  flex: 0 0 67.333%;
+  display: flex;
+  justify-content: space-between;
+  color: #121212;
```
Comment (e71az) [inline src/assets/Css/App.css:120]: You created a class for this value but forgot to use it in 3 cases.

File: src/assets/Css/App.css [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+.divider {
+  width: 0.125rem;
+  height: 4.375rem;
+  margin: 1.125rem 3.688rem 0 0;
+  border: 1px solid #e8e8e8;
+}
+
+.book-chapter {
+  flex: 0 0 32.667%;
+  color: #121212;
```
Comment (e71az) [inline src/assets/Css/App.css:244]: You created a class for this value but forgot to use it in 3 cases.

File: src/assets/Css/App.css [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+  font: bold 1.375rem "Roboto Slab", serif;
+  letter-spacing: -0.2px;
+}
+
+.author {
+  min-width: 6.688rem;
+  height: 1.188rem;
+  margin-top: 0.25rem;
+  margin-bottom: 0;
+  font-family: "Roboto Slab", serif;
```
Comment (e71az) [inline src/assets/Css/App.css:150]: You are repeating this Roboto value 4 times. (Excluding the parts where you are applying a special font size).\

File: src/assets/Css/App.css [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+}
+
+.header a {
+  min-width: 15rem;
+  height: 2.313rem;
+  margin: 0.375rem 2.938rem 0.125rem 0;
+  color: #0290ff;
+  font: bold 1.875rem "Montserrat", sans-serif;
+  text-decoration: none;
+  letter-spacing: normal;
```
Comment (e71az) [inline src/assets/Css/App.css:35]: This is repeated 10 times.

**Review by e71az — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️

Hello @george-swift,

Great job with the styling part of this project, your page is exact as the given design, however there is one important aspect that you need to improve before moving on:

- [ ] In the stylesheets of the app, make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright)
  - no repeating of the same rule

Currently, the problem that I can see is that you are repeating a lot of CSS rules in multiple elements for example: 
``` CSS
  #section-divider {
    margin-left: 3.09rem;
    margin-right: 3.09rem;
  }

  .bookform,
  #form-title {
    min-width: 0;
    margin-left: 3.09rem;
    margin-right: 3.09rem;
  }
```

Could be improved like this:

``` CSS

  #section-divider,
  .bookform,
  #form-title {
    margin-left: 3.09rem;
    margin-right: 3.09rem;
  }

  .bookform,
  #form-title {
    min-width: 0;
  }
```

You have multiple cases like this that could be reduced, please do so. I will mark some of them in this pull request comments section but please improve all the cases that you find out.

What I mean is that you can create new general classes where you can place all these repeated attributes and then just modify the other ones with specific changes.

[OPTIONAL]:

- [ ] Please get rid of all your ID selectors if this doesn't affect the react part of this project, if it does, ignore this. [Article about ids vs classes](https://dev.to/clairecodes/reasons-not-to-use-ids-in-css-4ni4)

Please correct those issues and send another review request, good luck, and well done!

**Review by trekab — APPROVED** [review]
### Approved! :tada: 
Nicely done, @george-swift! It is commendable that you put in the extra effort to make your app responsive! :+1: Keep up the good work.

_Happy coding!_
__[Treasure Kabareebe](https://trekab.github.io)__

**Pull request merged**